### PR TITLE
[base] In Language Allow to set first day of week

### PR DESCRIPTION
**Pull request by jat-odoo**
Description of the issue/feature this PR addresses:
* Task: https://www.odoo.com/web#id=34003&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
* Pad: https://pad.odoo.com/p/r.6542ffc7d26662edee490764b0df5e97

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by jat-odoo** [comment]
LGTM

**Comment by fmdl** [comment]
@jat-odoo I think it is better to put the configuration of `week_start` in user preference.

Because if you have a company in Dubai for exemple, Odoo is in english (because every body speak in english), but the week start on sunday.

File: odoo/addons/base/models/res_lang.py [hunk 1]
```diff
@@ -209,6 +217,12 @@ def get_installed(self):
     @api.model
     def create(self, vals):
         self.clear_caches()
+        if not vals.get('week_start'):
+            try:
+                locale = babel.Locale(vals['code'])
+            except babel.core.UnknownLocaleError:
+                locale = babel.Locale('en_US')
+            vals.update({'week_start': locale.first_week_day + 1})  # babel.Locale() starts indexing from 0
```
Comment (rim-odoo) [inline odoo/addons/base/models/res_lang.py]: Why not implementing this with a `default=` function on the field instead of overriding the `create()`?

Comment (suh-odoo) [reply odoo/addons/base/models/res_lang.py]: @rim-odoo 

we did not get current language code on default function that is only available on create method.

Comment (rim-odoo) [reply odoo/addons/base/models/res_lang.py]: seems legit indeed, keep the code as is

File: odoo/addons/base/models/res_lang.py [hunk 2]
```diff
@@ -33,6 +34,13 @@ class Lang(models.Model):
     direction = fields.Selection([('ltr', 'Left-to-Right'), ('rtl', 'Right-to-Left')], required=True, default='ltr')
     date_format = fields.Char(string='Date Format', required=True, default=DEFAULT_DATE_FORMAT)
     time_format = fields.Char(string='Time Format', required=True, default=DEFAULT_TIME_FORMAT)
+    week_start = fields.Selection([(1, 'Monday'),
+                                   (2, 'Tuesday'),
+                                   (3, 'Wednesday'),
+                                   (4, 'Thursday'),
+                                   (5, 'Friday'),
+                                   (6, 'Saturday'),
+                                   (7, 'Sunday')], string='First Day of Week')
```
Comment (rim-odoo) [inline odoo/addons/base/models/res_lang.py]: required

File: odoo/addons/base/models/res_lang.py (hunk 1 above)
Comment (rim-odoo) [reply odoo/addons/base/models/res_lang.py]: can't this be done with a onchange on `code` ?

Comment (suh-odoo) [reply odoo/addons/base/models/res_lang.py]: @rim-odoo ,

onchange not triggered on demo data record.

Comment (rim-odoo) [reply odoo/addons/base/models/res_lang.py]: the update the data ?

File: addons/web/static/src/js/views/calendar/calendar_model.js [hunk 3]
```diff
@@ -390,7 +390,7 @@ return AbstractModel.extend({
             monthNamesShort: moment.monthsShort(),
             dayNames: moment.weekdays(),
             dayNamesShort: moment.weekdaysShort(),
-            firstDay: moment().startOf('week').isoWeekday(),
+            firstDay: _t.database.parameters.week_start || moment().startOf('week').isoWeekday(),
```
Comment (Xavier-Do) [inline addons/web/static/src/js/views/calendar/calendar_model.js]: Is it useful to keep the moment() part? did you find a case where week_start was not defined?

Comment (suh-odoo) [reply addons/web/static/src/js/views/calendar/calendar_model.js]: You are right after adding the damo data i just forgot to remove this code.

**Pull request merged**